app/views/layouts/application.html.erb

```diff
@@ -12,8 +12,22 @@
     <nav class = "topnav">
       <%= link_to "All Merchants", "/merchants"%>
       <%= link_to "All Items", "/items"%>
-      <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-      <%= link_to "Register", "/register" %>
+      <% if !current_admin? %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+      <% end %>
+      <% if current_user %>
+          <%= link_to "Log out", "/logout", method: :delete %>
+      <% else %>
+        <%= link_to "Register", "/register" %>
+        <%= link_to "Login", "/login" %>
+      <% end %>
+      <% if current_merchant? %>
```

Comment:
I wonder if this is the best way to identify merchants and admins in here. Possible refactor possiblity later on.